src/pyswi/swi_ts/swi_ts.py

```diff
@@ -95,33 +95,33 @@ def swi_error_prop(ssm, t_value, t_noise, swi_error, gain_in=None, nan=-9999.):
 
     for i in range(len_ssm):
         while j < len_ssm and ssm['sm_jd'][j] <= ssm['sm_jd'][i]:
-            if ssm['sm'][j] != nan or ssm['sm'][j] == np.nan:
+            if ssm['sm'][j] != nan:
```

Comment:
it looks like the old version was able to handle np.nan, but the new one doesnt?

Reply:
more like "the old version tried to handle np.nan but did it wrong anyway". If anything it should be `np.isnan(ssm['sm'][j])` not `ssm['sm'][j] == np.nan` I guess.

Reply:
Old version initiated calculation for ssm = np.nan which isn't right.
There are no np.nan in input data, theres a 'nan' value param, e.g. -9999. in cci/c3s. That statement was obsolete anyway because np.nan != nan.

Reply:
So this is all correct

Reply:
ok, but wouldn't it make sense to handle np.nan too? We can do this separately if you want, just wondering.

Reply:
will write a test

Reply:
There should not be both a nan and np.nan in any input. 
If input has np.nan then nan=np.nan

Reply:
thanks. I think the "normal" case is not that there is a specific fill value but np.nan. So users would have to first replace np.NaNs with fill values that pyswi expects. And simply setting `nan=np.nan` wouldnt work as you know.

Reply:
But as I said, I can look at this in a separate PR if you don't want to spend time on this.

Reply:
in that case it should be:
 if ssm['sm'][j] != nan or np.isnan(ssm['sm'][j]) == False:
or
 if ssm['sm'][j] != nan and np.isinf(ssm['sm'][j]):
??

Reply:
I think `ssm['sm'][j] != nan or ~np.isnan(ssm['sm'][j])` should work